externals/kyuubi-hive-sql-engine/src/main/scala/org/apache/kyuubi/engine/hive/HiveSQLEngine.scala

```diff
@@ -100,6 +100,9 @@ object HiveSQLEngine extends Logging {
     EventBus.post(event)
     Utils.addShutdownHook(() => {
       engine.getServices.foreach(_.stop())
+      val event = HiveEngineEvent(engine)
+        .copy(state = ServiceState.STOPPED, endTime = System.currentTimeMillis())
+      EventBus.post(event)
```

Comment:
it's unreliable to do some io work in shutdown hook, can we do this in `stopServer` ?

Reply:
It will be not be executed in stopServer.

Reply:
I mean, can we move this before exit system ?

```scala
  override protected def stopServer(): Unit = {
    // #2351
    // https://issues.apache.org/jira/browse/HIVE-23164
    // Server is not properly terminated because of non-daemon threads
    System.exit(0)
  }
```

Reply:
done